kubernetes/test-infra/gke/main.tf

```diff
@@ -36,7 +40,7 @@ resource "google_service_account" "default" {
 
 resource "google_container_cluster" "primary" {
   provider           = google-beta
-  name               = "tf-acc-test-${random_id.cluster_name.hex}"
+  name               = var.cluster_name != "" ? var.cluster_name : "tf-acc-test-${random_id.cluster_name.hex}"
```

Comment:
This change is not related to the fix directly, however, I found this useful to configure the cluster name for manual testing purposes while working on this issue.